cloudevents/pydantic/event.py

```diff
@@ -0,0 +1,259 @@
+            " changes to the specification to be made without changing this property's"
+            " value in the serialization. Note: for 'release candidate' releases a"
+            " suffix might be used for testing purposes."
+        ),
+    )
+    time: typing.Optional[datetime.datetime] = pydantic.Field(
+        default_factory=attribute.default_time_selection_algorithm,
+        title="Occurrence Time",
+        description=(
+            " Timestamp of when the occurrence happened. If the time of the occurrence"
+            " cannot be determined then this attribute MAY be set to some other time"
+            " (such as the current time) by the CloudEvents producer, however all"
+            " producers for the same source MUST be consistent in this respect. In"
+            " other words, either they all use the actual time of the occurrence or"
+            " they all use the same algorithm to determine the value used."
+        ),
+    )
+
+    subject: typing.Optional[str] = pydantic.Field(
+        title="Event Subject",
+        description=(
+            "This describes the subject of the event in the context of the event"
+            " producer (identified by source). In publish-subscribe scenarios, a"
+            " subscriber will typically subscribe to events emitted by a source, but"
+            " the source identifier alone might not be sufficient as a qualifier for"
+            " any specific event if the source context has internal"
+            " sub-structure.\n"
+            "\n"
+            "Identifying the subject of the event in context"
+            " metadata (opposed to only in the data payload) is particularly helpful in"
+            " generic subscription filtering scenarios where middleware is unable to"
+            " interpret the data content. In the above example, the subscriber might"
+            " only be interested in blobs with names ending with '.jpg' or '.jpeg' and"
+            " the subject attribute allows for constructing a simple and efficient"
+            " string-suffix filter for that subset of events."
+        ),
+    )
+    datacontenttype: typing.Optional[str] = pydantic.Field(
+        title="Event Data Content Type",
+        description=(
+            "Content type of data value. This attribute enables data to carry any type"
+            " of content, whereby format and encoding might differ from that of the"
+            " chosen event format."
+        ),
+    )
+    dataschema: typing.Optional[str] = pydantic.Field(
+        title="Event Data Schema",
+        description=(
+            "Identifies the schema that data adheres to. "
+            "Incompatible changes to the schema SHOULD be reflected by a different URI"
+        ),
+    )
+
+    def __init__(
+        self,
+        attributes: typing.Optional[typing.Dict[str, typing.Any]] = None,
+        data: typing.Optional[typing.Any] = None,
+        **kwargs
```

Comment:
So we're basically saying that we're supporting either `kwargs` or `attributes`. Why don't we make `attibutes` keyword-only arg then?
```suggestion
    def __init__(
        self,
        *,
        attributes: typing.Optional[typing.Dict[str, typing.Any]] = None,
        data: typing.Optional[typing.Any] = None,
        **kwargs
```

Reply:
@sasha-tkachev PTAL at this one.

Reply:
We're gonna keep it as is to be consistent with the standard implementation.